lighthouse-core/gather/gather-runner.js

```diff
@@ -382,17 +389,14 @@ class GatherRunner {
       }
     }
 
-    // Take only unique LighthouseRunWarnings.
-    baseArtifacts.LighthouseRunWarnings = Array.from(new Set(baseArtifacts.LighthouseRunWarnings));
-
-    // Take the timing entries we've gathered so far.
-    baseArtifacts.Timing = log.getTimeEntries();
-
-    // TODO(bckenny): correct Partial<LH.GathererArtifacts> at this point to drop cast.
-    return /** @type {LH.Artifacts} */ ({...baseArtifacts, ...gathererArtifacts});
+    return {
+      artifacts: gathererArtifacts,
```

Comment:
>> feels a bit weird that this isnt just returing the partial artifacts then...
> oh I see it's so later on it matches the signature of the other method :)

(I know you're already well aware of all this, @patrickhulce, but for anyone else reading :)

it's not my favorite thing ever, but we really have three finished-pass states
- regular artifacts returned
- fatal exception thrown (e.g. javascript syntax error, LH logic error)
- `pageLoadError`, when the page load failed but LH is doing fine (bad URL, bad cert, etc)

The first one is handled by the existing returned artifacts interface, the second is handled by normal exceptions, but the third we don't want to handle with exceptions because it's not really an exceptional state for Lighthouse.

That's been fine before because we can just handle the `pageLoadError` within a pass by erroring out all the artifacts from that pass, but now we want the error information higher up, so we need to pass back artifacts and the `pageLoadError`. There are other ways of doing this, but really we're just talking about a multi-value return, so let's just do a multi-value return :)